src/sql/connection/connection.py

```diff
@@ -527,6 +529,13 @@ def connection(self):
         """Returns the SQLAlchemy connection object"""
         return self._connection_sqlalchemy
 
+    def close(self):
+        super().close()
+
+        # NOTE: in SQLAlchemy 2.x, we need to call engine.dispose() to completely
+        # close the connection, calling connection.close() is not enough
+        self.connection.engine.dispose()
```

Comment:
should we check if it's sqlalchemy 2 before running this command?

Reply:
not needed, sqlalchemy 1.x also has this method so nothing breaks